catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -422,10 +422,18 @@ public CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
           return queryResponse;
         };
 
-    List<Result> results =
-        ResultIterable.resultIterable(queryFunction, request, cqlRequest.getCount())
-            .stream()
-            .collect(Collectors.toList());
+    List<Result> results;
+    if (cqlRequest.getCount() == 0) {
+      results =
+          ResultIterable.resultIterable(queryFunction, request)
+              .stream()
+              .collect(Collectors.toList());
+    } else {
+      results =
+          ResultIterable.resultIterable(queryFunction, request, cqlRequest.getCount())
+              .stream()
+              .collect(Collectors.toList());
```

Comment:
✏️ If you don't like duplicating the stream and collect you can always, but either ways are pretty similar. 
```java
ResultIterable iterable;
if (cqlRequest.getCount() == 0) {
  iterable = resultIterable(queryFunction, request);
} else {
  iterable = resultIterable(queryFunction, request, cqlRequest.getCount())
}

List<Result> results = iterable.stream().collect(Collectors.toList());
```